src/sipreg/reg.c

```diff
@@ -618,10 +611,13 @@ int sipreg_set_fbregint(struct sipreg *reg, uint32_t fbregint)
  */
 void sipreg_set_srcport(struct sipreg *reg, uint16_t srcport)
 {
-	if (!reg)
+	if (!reg || !reg->dlg)
```

Comment:
Is setting the srcport only relevant for dlg? Or is there a use-case without it too?

Reply:
Currently not. We could add a setter `void sip_request_set_srcport(...)`.

For now I couldn't find any call to plain `sip_request()` in re or baresip. And no call to `sip_request_alloc()` followed by `sip_request_send()`.

For `sipreg` I think this is a correct condition, because it always has a dialog.

Reply:
Just saw that `sip_request_alloc()` is not in the API. So a setter `sip_request_set_srcport()` doesn't make sense.